x/twap/store.go

```diff
@@ -94,23 +94,29 @@ func (k Keeper) pruneRecordsBeforeTimeButNewest(ctx sdk.Context, lastKeptTime ti
 	seenPoolAssetTriplets := map[uniqueTriplet]struct{}{}
 
 	for ; iter.Valid(); iter.Next() {
-		twapToRemove, err := types.ParseTwapFromBz(iter.Value())
+		timeIndexKey := iter.Key()
+		timeS, poolId, asset0, asset1, err := types.ParseFieldsFromHistoricalTimeKey(timeIndexKey)
 		if err != nil {
 			return err
 		}
 
 		poolKey := uniqueTriplet{
-			poolId: twapToRemove.PoolId,
-			asset0: twapToRemove.Asset0Denom,
-			asset1: twapToRemove.Asset1Denom,
+			poolId,
+			asset0,
+			asset1,
 		}
 		_, hasSeenPoolRecord := seenPoolAssetTriplets[poolKey]
 		if !hasSeenPoolRecord {
 			seenPoolAssetTriplets[poolKey] = struct{}{}
 			continue
 		}
 
-		k.DeleteHistoricalRecord(ctx, twapToRemove)
```

Comment:
This method now seems unused, can we delete the method as well?

Reply:
Hrmm, seems like a useful helper / reference implementation tho